src/rust/config/src/debug.rs

```diff
@@ -25,17 +26,30 @@ fn log_nbuf() -> usize {
 // struct definitions
 #[derive(Serialize, Deserialize, Debug)]
 pub struct DebugConfig {
+    #[serde(with = "LevelDef")]
     #[serde(default = "log_level")]
-    log_level: usize,
+    log_level: Level,
     #[serde(default = "log_file")]
     log_file: Option<String>,
     #[serde(default = "log_nbuf")]
     log_nbuf: usize,
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "lowercase")]
+#[serde(remote = "Level")]
+#[serde(deny_unknown_fields)]
+enum LevelDef {
```

Comment:
Yeah - we can do something better here once we have a more full-featured logger implementation in rustcommon. I have something started to get us to production-ready logging. I think what we'll do is invent some 'pseudo-levels' that help round out the levels defined by the `log` crate's macro system: https://docs.rs/log

It's important to consider that for libraries, we should really only use those standard levels. I will re-audit this PR for my own use of a rustcommon specific `fatal!` level which is just a convenience macro of `error!` + exit - I think it's fine for use in a binary crate where we dictate the dependencies, but for libraries we should use the levels defined by `log` if we ever want others to use our libraries.

We also have a somewhat unique aspect here where the binary crate in twemcache-rs overrides the `panic!` hook to log at `error!` level, print out a backtrace (currently on stdout), and then exit with error code. This applies to `panic!`, `expect()`, `unwrap()`, and the various `assert`s. While we should try to limit all of those in library code (preferring to let a caller handle a `Result<>` type with a useful `Error`) - in some cases in segcache we may want to leave `panic!` and friends in place (likely better to panic than to segfault if corruption were to happen). Without overriding the panic hook, typically just the `panic!`ing thread will terminate, for a server process like this, it's better to bring the whole thing down after getting a backtrace.

For possible 'pseudo-levels' in a more advanced logger, we can control whether we want to include trace or debug logging from external crates - for instance, the mio trace log is very heavy and we likely don't want to include it by default at an overall 'trace' level. We could add something like 'very trace' which doesn't filter out external crates. Similarly for debug, we probably care more about debug messaging from our own code rather than other code we depend on unless we opt into it with 'very debug' or similar.

Will discuss further with you when the time comes to make decisions about this, for now the logging is not filtered by the crate that produced the log message.